audio/audio_utils.R

```diff
@@ -0,0 +1,231 @@
+exponentialFun <- function(k, a, b) {return(round(a*exp(-b*k)))}
+
+make_wav <- function(number_of_speakers = 2, audio_annotation, max_duration, balanced = T, outfolder){
+  
+  n_speakers <- sample(unique(audio_annotation$name), number_of_speakers)
+  sub_n_speakers <- audio_annotation[audio_annotation$name %in% n_speakers,]
+  
+  wav_file <- NULL
+  
+  timeDistribution <- rep(max_duration, number_of_speakers)
+  
+  if(balanced == F) {
+    k <- c(0:number_of_speakers); a = max_duration; b = .6
+    
+    timeDistribution <- NULL
+    for (i in 1:length(k)){
+      timeDistribution[i]<- exponentialFun(k[i], a, b)
+    }
+    
+  }
+  table_duration <- aggregate(tdur ~ name, data = sub_n_speakers, sum)
+  ordered_speakers <- table_duration$name
+  
+  for (n in 1:length(n_speakers)){
+    speaker_subset <- sub_n_speakers[sub_n_speakers$name == ordered_speakers[n],]
+    speaker_tdur <- speaker_subset$tdur
+    
+    max_duration_ <- timeDistribution[n]
+    
+    # select rows based on max duration
+    idx <- which(cumsum(speaker_tdur) <= max_duration_)
+    
+    if(length(idx) == 0){
+      wav_file <- dplyr::bind_rows(wav_file, speaker_subset[1,])
+    }
+    
+    wav_file <- dplyr::bind_rows(wav_file, speaker_subset[idx,])
+    wav_file$max_duration <- max_duration
+    wav_file$n_speaker <- number_of_speakers
+    wav_file$balanced <- balanced
+    
+    write.csv(wav_file, file.path(outfolder, paste0("speaker_", number_of_speakers, "_duration_", max_duration, "_",ifelse(balanced == T, "balanced","unbalanced"), ".csv")))
```

Comment:
```suggestion
    write.csv(wav_file, file.path(outfolder, paste0("speaker_", number_of_speakers, "_duration_", max_duration, "_",ifelse(balanced, "balanced","unbalanced"), ".csv")))
```

For logical variables, you don't need to test if they are equal to `TRUE`. This occurs a few times in this file.

Another suggestion for good practice: Instead of `T` and `F`, write out `TRUE` and `FALSE`. This is a bit more tedious to type but prevents (hard to find) bugs when you assign values to `T` and `F`, e.g., if you do `T = 0` beforehand.

Reply:
what do you mean 'For logical variables, you don't need to test if they are equal to TRUE'? You mean that I can use binary numbers? e.g., TRUE/1 -- FALSE/0? 

for the second suggestion, I agree that this is less prone to bugs -- I'm adding it

Reply:
I mean that instead of writing `ifelse(balanced == TRUE, ..., ...)`, you can write `ifelse(balanced, ..., ...)`. You don't need to explicitly test whether logical variables are equal to `TRUE`, this will be alway done implicitly.

Reply:
ahhhhh! This is great, yes I'll do that then. Thanks for the tip!